### PR TITLE
Actually call to create a default annotation on next or on done

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team. If PR is related to design, please request review from `@beckyrother` in addition._ 

Package: lib-classifier

Describe your changes:
Right now, staging 400 errors because the default annotation isn't actually being created when you click next or done. This was missed by me in review of #1228. I've fixed this by removing the now cruft `createDefaultAnnotation` and calling `addAnnotation` directly. I've added some tests to check that default annotations are actually created for the TaskNav container. 

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
I think you can call `task.createAnnotation()` to create the missing annotation.

**Comment by srallen** [comment]
> I think you can call task.createAnnotation() to create the missing annotation.

My tests start failing because they do not get added to the classification annotation if I call `task.createAnnotation()` directly in the TaskNavContainer. Calling `addAnnotation` ensures it's being created and added to the annotations. I'm not sure if I'm doing something wrong with this suggestion but the current solution is working.

**Comment by eatyourgreens** [comment]
You’re right. Tasks create annotations but aren’t responsible for storing them, because they might be creating them for drawing subtasks.

If the annotation value is optional in `addAnnotation`, it might be worth changing the signature of that action to `addAnnotation(task, value)` so that it’s easier to call with just the task.

**Comment by eatyourgreens** [comment]
`task.updateAnnotation(value)` is what I was trying to think of. It will create an annotation on the classification if it doesn't exist. https://github.com/zooniverse/front-end-monorepo/blob/733d7155a8764b530c962eb7e2717147ff830517/packages/lib-classifier/src/store/tasks/Task.js#L22-L25

File: packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/TaskNavButtonsContainer.spec.js [hunk 1]
```diff
@@ -4,28 +4,36 @@ import { observable } from 'mobx'
 import { expect } from 'chai'
 import sinon from 'sinon'
 import TaskNavButtonsContainer from './TaskNavButtonsContainer'
+import ClassificationStore from '../../../../../../../../store/ClassificationStore'
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/TaskNavButtonsContainer.spec.js]: Are we still waiting to merge the PR that aliases these to `@store/ClassificationStore` etc?

File: packages/lib-classifier/src/store/ClassificationStore.js [hunk 2]
```diff
@@ -116,7 +116,9 @@ const ClassificationStore = types
           const annotation = classification.annotations.get(task.taskKey) || task.createAnnotation()
           // new annotations must be added to this store before we can modify them
           classification.annotations.put(annotation)
-          annotation.value = annotationValue
+          if (annotationValue) {
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/ClassificationStore.js]: Outside the scope of this PR, but maybe `annotation.update(annotationValue)` and then each annotation is responsible for validating the type and value of `annotationValue`.

File: packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/TaskNavButtonsContainer.spec.js [hunk 3]
```diff
@@ -214,24 +188,29 @@ describe('TaskNavButtonsContainer', function () {
...
     })
 
-    it('should prevent the event default', function () {
+    it('should create a default annotation for each task if there is not an annotation for that task', function () {
       wrapper.instance().onSubmit({ preventDefault: preventDefaultSpy })
-      expect(preventDefaultSpy).to.have.been.calledOnce()
+      const classification = classificationStore.active.toJSON()
+
+      tasks.forEach((task) => {
+        expect(addAnnotationSpy.withArgs(null, task)).to.have.been.calledOnce()
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/TaskNavButtonsContainer.spec.js]: Outside the scope of this PR, but a better test here would be to expect that the annotation for this task, stored on the classification, is the default annotation.

Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/TaskNavButtonsContainer.spec.js]: Nevermind, that's the next check. This expectation can be deleted. We don't care how the default annotation is implemented, only that it's created. We want to test behaviour, not implementation.

**Comment by eatyourgreens** [comment]
Testing this, I've discovered that tests all pass even if  the `steps` prop is an empty array. I'm not sure what's going on there, but `should call props.selectStep when there is a previous step` passes even without any steps.

**Comment by eatyourgreens** [comment]
@srallen the bugfix looks good to me. I've updated it to use `task.updateAnnotation()` and added a workflow step store to the tests.

**Comment by eatyourgreens** [comment]
PFE creates the default annotation when a volunteer begins a task. https://github.com/zooniverse/Panoptes-Front-End/blob/6d0656ddfda20bd83d2e7d8bfd63294b5bb8b1a5/app/classifier/task-nav.jsx#L23-L40

Maybe the classifier should do that too, rather than creating the annotation when the task is complete?

**Comment by eatyourgreens** [comment]
The fix seems to work, but the tests aren't generating the default annotation, yet still pass.

**Comment by srallen** [comment]
> Maybe the classifier should do that too, rather than creating the annotation when the task is complete?

There was a reason why I didn't do this, but it's not immediately coming to mind.

**Comment by eatyourgreens** [comment]
I started down the route of mocking a `WorkflowStepStore` for this, since the task nav buttons depend heavily on it, but gave up. It seemed like it would be a good idea to mock the initial step store state in order to generate initial props for some of the tests, rather than mock props which might not reflect a real store.

**Comment by srallen** [comment]
I think refactoring the step portion of these tests is out of scope.

**Pull request merged**